lib/perl/Genome/Site/TGI/Extension/Sys.pm

```diff
@@ -69,6 +71,82 @@ sub get_lsf_job_status {
     return $status;
 }
 
+sub bsub_and_wait_for_completion {
+    my($class, %bsub_args) = @_;
+
+    my $special_args;
+    ($special_args, %bsub_args) = _bsub_and_wait_for_completion__validate_args(%bsub_args);
+    my($commands, $on_submit_cb, $on_complete_cb) = @$special_args{'commands','on_submit','on_complete'};
+
```

Comment:
It would be clearer if these were all separate key value pairs passed in as params. Maybe they were too many places to update to the new functionality?